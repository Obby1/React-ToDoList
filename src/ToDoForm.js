import React, { useState } from 'react';
import { v4 as uuid } from "uuid";

function ToDoForm({ addToDo, ...props }) {
  const INITIAL_STATE = { text: '', completed: false};
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((fData) => ({ ...fData, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addToDo({ ...formData, id: uuid() });
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit} {...props}>
      <label htmlFor="text">Text:</label>
      <input id="text" name="text" value={formData.text} onChange={handleChange} />
      <button>Add To Do</button>
    </form>
  );
}

export default ToDoForm;
