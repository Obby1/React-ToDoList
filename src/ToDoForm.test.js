import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ToDoForm from './ToDoForm';



it('renders ToDoForm without crashing', () => {
    render(<ToDoForm addToDo={() => {}} />);
  });
  
it('matches ToDoForm snapshot', () => {
    const { asFragment } = render(<ToDoForm addToDo={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
// core functinality tests for updating, removing, editing tests located in ToDoList.test.js
test('calls addToDo prop when form is submitted', () => {
    const addToDo = jest.fn();
    const { getByLabelText, getByText } = render(<ToDoForm addToDo={addToDo} />);
    const textInput = getByLabelText('Text:');
    const addButton = getByText('Add To Do');
  
    fireEvent.change(textInput, { target: { value: 'Test ToDo' } });
    fireEvent.click(addButton);
  
    expect(addToDo).toHaveBeenCalled();
  });
  

