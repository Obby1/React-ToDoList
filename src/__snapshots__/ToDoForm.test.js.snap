// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches ToDoForm snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Text:
    </label>
    <input
      id="text"
      name="text"
      value=""
    />
    <button>
      Add To Do
    </button>
  </form>
</DocumentFragment>
`;
